packages/core/src/extensions/typenames.ts

```diff
@@ -0,0 +1,19 @@
+import { Extension } from '../Extension.js'
+
+export const Typenames = Extension.create({
+  name: 'typenames',
+
+  addGlobalAttributes() {
+    return this.extensions.filter(extension => {
+      return extension.name !== 'text' && extension.name !== 'doc'
+    }).map(extension => ({
+      types: [extension.name],
+      attributes: {
+        'data-tiptap': {
+          default: extension.name,
+          rendered: true,
+        },
```

Comment:
Does this get stored to the Tiptap JSON? Kind of redundant if it does

Reply:
Don't think those are added to the Tiptap JSON.